src/lazy/encoder/writer.rs

```diff
@@ -846,21 +939,30 @@ impl<V: ValueWriter> SequenceWriter for ApplicationSExpWriter<'_, V> {
 }
 
 pub struct ApplicationEExpWriter<'value, V: ValueWriter> {
-    encoding: &'value mut WriterContext,
+    symbols: &'value mut WriterSymbolTable,
+    macros: &'value WriterMacroTable,
     raw_eexp_writer: V::EExpWriter,
     value_writer_config: ValueWriterConfig,
+    // TODO: these are now available but not yet used.
+    _invoked_macro: &'value MacroDef,
+    _param_index: usize,
```

Comment:
🪧 The next PR will add validation for each argument-writing method call that happens next.